account_banking_pain_base/models/res_bank.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import re
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class ResBank(models.Model):
+    _inherit = 'res.bank'
+
+    @api.multi
+    @api.constrains('bic')
+    def _check_bic(self):
+        """
+        This method strengthens the constraint on the BIC of the bank account
+        (The account_payment_order module already checks the length in the
+         check_bic_length method).
+        :raise: ValidationError if the BIC doesn't respect the regex of the
+        SEPA pain schemas.
+        """
+        bic_re = r"[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}"
```

Comment:
A few things, from important to nitpicking:
- add a `$` at the end, to make sure longer strings do not match
- use `re.compile` to precompile the regexp as a module-level constant
- `{0,1}` is more commonly written `?`
- `{6,6}` can also be written `{6}`